rusty_pizza_server/src/order_model/meals.rs

```diff
@@ -106,4 +128,53 @@ mod tests {
             }
         );
     }
+
+    #[rstest(prices, expected_total,
+        case(vec![Money::new(2, 25), Money::new(5, 50), Money::new(7, 33)], Money::new(15, 08)),
+        case(vec![Money::new(3, 50), Money::new(4, 42)], Money::new(7, 92)),
+    )]
+    fn total_price_is_calculated_correctly(prices: Vec<Money>, expected_total: Money) {
+        //Given
+        let user = Rc::new(User::new(String::from("Peter")));
+        let mut meals = Meals::new(user.clone());
+        let mut meal_factory = MealFactory::new();
+
+        for price in prices.into_iter() {
+            let meal =
+                meal_factory.create_meal(String::from("XX"), String::from("something"), price);
```

Comment:
The `MealFactory` isn't actually needed here. You could use the constructor `Meal::new`. In my opinion this would simplify the test.

Reply:
I don´t think that the constructor would simplify the test. The test create more than one meal, so if use the constructor I have to manage the ids by my self. But the ids are not important for the tests `total_price_is_calculated_correctly` or `change_is_calculated_correctly`